### DISCUSSION TITLE
How to set permission mode on unix socket in `--rpclisten=`

### DISCUSSION THREAD

**Discussion by dpc** [lightningnetwork/lnd#7521] [Q&A] [answered]
I'm using `--rpclisten=unix://...`but would like to give access to it to other users, however it seems like it's only writeable by the owner user:

```
srwxr-xr-x 1 lnd lnd 0 Mar 17 17:43 /var/lib/lnd/rpc
```

so `lncli` is getting permission error when trying to connect to it.

After I used `chmod ugo+w /var/lib/lnd/rpc` everything is working again.

I guess I'm looking for something like `--rpc-file-mode` from https://lightning.readthedocs.io/lightningd-config.5.html

**Comment by guggero** [comment, marked as answer]
It sounds like we would need to add such a flag as well and call `os.Chmod()` explicitly after creating the socket: https://github.com/golang/go/issues/11822

As a workaround you could use [`lndinit wait-ready`](https://github.com/lightninglabs/lndinit#wait-ready) and then `chmod` the socket yourself:
```shell

lndinit wait-ready && chmod 777 /var/lib/lnd/rpc

```

Can you create a feature request for the `--rpc-file-mode` equivalent flag please?
